internal/services/mssql/mssql_database_resource.go

```diff
@@ -759,6 +799,20 @@ func resourceMsSqlDatabaseRead(d *pluginsdk.ResourceData, meta interface{}) erro
 		return fmt.Errorf("setting `geo_backup_enabled`: %+v", err)
 	}
 
+	if features.ThreePointOh() {
+		tde, err := transparentEncryptionClient.Get(ctx, id.ResourceGroup, id.ServerName, id.Name)
+		if err != nil {
+			return fmt.Errorf("while retrieving Transparent Data Encryption status of %q: %+v", id.String(), err)
+		}
+		tdeStatus := false
+		if tde.TransparentDataEncryptionProperties.Status == sql.TransparentDataEncryptionStatusEnabled {
+			tdeStatus = true
+		}
+		if err := d.Set("transparent_data_encryption_enabled", tdeStatus); err != nil {
+			return fmt.Errorf("setting transparent_data_encryption: %+v", err)
+		}
```

Comment:
this is a bool so there's no need to error check those?

```suggestion
		d.Set("transparent_data_encryption_enabled", tdeStatus)
```